src/service.rs

```diff
@@ -57,6 +57,22 @@ pub struct BlockInfo<Types: NodeType> {
     pub offered_fee: u64,
 }
 
+#[derive(Debug)]
+pub struct BuiderStatesInfo<Types: NodeType> {
+    pub vid_commitments: Vec<VidCommitment>,
+    pub block_ids: Vec<(VidCommitment, BuilderCommitment, Types::Time)>,
+}
+
+// impl default for the builderstateinfo
+impl<Types: NodeType> Default for BuiderStatesInfo<Types> {
```

Comment:
This should be derivable. If it's giving you trouble because `Types` doesn't implement `Default`, you can remove that requirement with derivative:

```rust
#[derive(Derivative)]
#[derivative(Default(bound = ""))]
struct BuilderStatesInfo<Types> { ... }
```

Reply:
https://github.com/EspressoSystems/hotshot-builder-core/pull/120/commits/4ed541c97fcdbb456d5c3d53d370316dad528fe3